src/BottomNavigation/BottomNavigationItem.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import shallowEqual from 'recompose/shallowEqual';
+import EnhancedButton from '../internal/EnhancedButton';
+
+function getStyles(props, context) {
+  const {selected} = props;
+
+  const {muiTheme} = context;
+  const {bottomNavigation} = muiTheme;
+
+  const color = selected ? bottomNavigation.selectedColor :
+                           bottomNavigation.unselectedColor;
+  const styles = {
+    root: {
+      transition: 'all 0.3s',
```

Comment:
> However, it is possible to pass an arbitrary element as an icon, which could have any number of properties that need transitioning.

Well, good point. But shouldn't it be the responsibility of users to add additional transitions when they are changing more properties that the component is expecting?